frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/highlightNodesAndEdges.ts

```diff
@@ -0,0 +1,127 @@
+import type { Edge, Node } from '@xyflow/react'
+import { type TableNodeType, isTableNode } from './TableNode'
+
+type SourceTableName = string
+type TargetTableName = string
+type EdgeMap = Map<SourceTableName, TargetTableName[]>
+
+const isActiveNode = (
+  activeTableName: string | undefined,
+  node: TableNodeType,
+): boolean => {
+  return node.data.table.name === activeTableName
+}
+
+const isActivelyRelatedNode = (
+  activeTableName: string | undefined,
+  edgeMap: EdgeMap,
+  node: TableNodeType,
+): boolean => {
+  if (!activeTableName) {
+    return false
+  }
+
+  return edgeMap.get(activeTableName)?.includes(node.data.table.name) ?? false
```

Comment:
By making a memo of `Map<SourceTableName, TargetTableName[]>` once, you can then determine whether they are related with a small amount of looping.
This should also work for the logic during hover. This will be handled later.